pkg/lightning/backend/kv/base.go

```diff
@@ -144,23 +144,25 @@ func NewBaseKVEncoder(config *encode.EncodingConfig) (*BaseKVEncoder, error) {
 
 	var autoRandomColID int64
 	autoIDFn := func(id int64) int64 { return id }
-	if meta.ContainsAutoRandomBits() {
-		col := common.GetAutoRandomColumn(meta)
-		autoRandomColID = col.ID
+	if !config.UseIdentityAutoRowID {
```

Comment:
Why here need add this condition check for non-autoRowID?

Reply:
on default, encoder will auto DO shard for `sharded auto row ID` and `auto random`, but during conflict resolution, we want the row ID passed in to be used as is.